lib/storage/src/content_manager/snapshots/recover.rs

```diff
@@ -278,6 +279,10 @@ async fn _do_recover_from_snapshot(
                             true,
                         )?;
                     }
+
+                    // `ShardTransfer` is only used during snapshot *shard transfer*.
+                    // It is only exposed in internal gRPC API and only used for *shard* snapshot recovery.
+                    SnapshotPriority::ShardTransfer => unreachable!(),
```

Comment:
Could you elaborate on why this branch is unreachable, while the other one in `src/common/snapshots.rs` is not?

Is this code only used in the REST API, while the branch in the other file is for the gRPC API?

Reply:
If the above is the case, I wonder if it would be better to just do `()` here as well. Maybe we don't want to panic.

Reply:
- there are no `recover` methods in public gRPC API
- `SnapshotPriority::ShardTransfer` is not exposed in HTTP API
- this particular function is only used for *collection* snapshot recovery
- and similar-ish function in `src/common/snapshots.rs` is only used for *shard* snapshot recovery
- so `SnapshotPriority::ShardTransfer` can't be propagated into *collection* snapshot recovery function
  - not present (at all) in gRPC API
  - unavailable in HTTP API
  - if this branch *is* reached, that's a logical error somewhere

Reply:
> this particular function is only used for collection snapshot recovery

That explains. Thanks! Sorry for missing that.